applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c

```diff
@@ -291,14 +291,25 @@ static void mqtt_thread_fn(void *arg1, void *arg2, void *arg3)
 			if ((fds.revents & POLLERR) == POLLERR) {
 				LOG_ERR("POLLERR");
 				mqtt_abort(&client);
+				err = -EIO;
+				break;
+			}
+			if ((fds.revents & POLLHUP) == POLLHUP) {
+				LOG_ERR("POLLHUP");
+				mqtt_abort(&client);
+				err = -ECONNRESET;
 				break;
 			}
 			if ((fds.revents & POLLNVAL) == POLLNVAL) {
 				LOG_ERR("POLLNVAL");
 				mqtt_abort(&client);
+				err = -ECONNABORTED;
 				break;
 			}
 		}
+		if (err) {
```

Comment:
Why would you exit the MQTT thread on a socket error? It should rather block on the semaphore, waiting for another connection (as it is implmented).

Reply:
Yes original logic is NEVER quit, but Coverity does not allow this behavior. We "break" for now and Larry will change to allow restart of thread later.

Reply:
> but Coverity does not allow this behavior

Ok, understood, a bit wierd from the Coverity side though...